docs/concepts/models.md

```diff
@@ -783,43 +783,114 @@ print(concrete_model(a=1, b=1))
 
     If you need to perform isinstance checks against parametrized generics, you can do this by subclassing the parametrized generic class. This looks like `class MyIntModel(MyGenericModel[int]): ...` and `isinstance(my_model, MyIntModel)`.
 
-If a Pydantic model is used in a `TypeVar` constraint, [`SerializeAsAny`](serialization.md#serializing-with-duck-typing) can be used to
-serialize it using the concrete model instead of the model `TypeVar` is bound to.
+If a Pydantic model is used in a `TypeVar` constraint or bound and the generic type is never parametrized then Pydantic will use the constraint or `TypeVar` default for validation but treat the value as `Any` in terms of serialization:
 
 ```py
-from typing import Generic, TypeVar
+from typing import Generic, Optional, TypeVar
 
-from pydantic import BaseModel, SerializeAsAny
+from pydantic import BaseModel
 
 
-class Model(BaseModel):
-    a: int = 42
+class ErrorDetails(BaseModel):
+    foo: str
 
 
-class DataModel(Model):
-    b: int = 2
-    c: int = 3
+ErrorDataT = TypeVar('ErrorDataT', bound=ErrorDetails)
+
+
+class Error(BaseModel, Generic[ErrorDataT]):
+    message: str
+    details: Optional[ErrorDataT]
+
+
+class MyErrorDetails(ErrorDetails):
+    bar: str
+
+
+# serialized as Any
+error = Error(
+    message='We just had an error',
+    details=MyErrorDetails(foo='var', bar='var2'),
+)
+assert error.model_dump() == {
+    'message': 'We just had an error',
+    'details': {
+        'foo': 'var',
+        'bar': 'var2',
+    },
+}
+
+# serialized using the concrete parametrization
+# note that `'bar': 'var2'` is missing
+error = Error[ErrorDetails](
+    message='We just had an error',
+    details=ErrorDetails(foo='var'),
+)
+assert error.model_dump() == {
+    'message': 'We just had an error',
+    'details': {
+        'foo': 'var',
+    },
+}
+```
 
+If you use a `default=...` for a `TypeVar` (available in Python >= 3.13 or via `typing-extensions`) the default value will be used for both validation and serialization if the type variable is not parametrized. You can override this behavior using `pydantic.SerializeAsAny`:
```

Comment:
@dmontagu I think this is perhaps the controversial part of this PR. My reasoning was that a bound and default are different semantically and this distinction led me to implement other behavior. This is also an opportunity to preserve the `SerializeAsAny` example that was here before. Let me know what you think.

Reply:
I agree with your interpretation of the semantics. (And in particular, this choice.)